### PR TITLE
Work around pyasn1's willingness to return endOfOctets in DER parsing

### PR DESCRIPTION
**Pull request by geofft**
See #1838 for discussion.

### CODE REVIEW

**Comment by reaperhulk** [comment]
add to whitelist

File: src/cryptography/hazmat/primitives/asymmetric/utils.py [hunk 1]
```diff
@@ -28,6 +29,9 @@ def decode_rfc6979_signature(signature):
         raise ValueError(
             "The signature contains bytes after the end of the ASN.1 sequence."
         )
+    if data == eoo.endOfOctets:
```
Comment (alex) [inline src/cryptography/hazmat/primitives/asymmetric/utils.py]: Can you add a comment about the fact that this occurs due to a PyASN1 bug, and can be removed in a hypothetical future where eoo is handled correctly?

Comment (geofft) [inline src/cryptography/hazmat/primitives/asymmetric/utils.py]: Done. It's harmless, so unless you'd prefer to include text about removing it, it seems more robust to keep it indefinitely. If you want to add something about removing it, I can add a comment after I file a bug (so we can link to the bug).

Comment (alex) [inline src/cryptography/hazmat/primitives/asymmetric/utils.py]: Hmm? I see you force pushed, but the contents looks like the same.

**Pull request merged**